lib/compositor_utils.ex

```diff
@@ -103,6 +112,32 @@ defmodule RecordingConverter.Compositor do
     }
   end
 
+  defp should_have_avatar?(
+         %{"origin" => origin} = track,
+         timestamp,
+         video_tracks_origin,
+         video_tracks_offset
+       ) do
+    origin not in video_tracks_origin and
+      longer_than_treshold?(track, timestamp) and
+      not has_video_in_threshold?(origin, video_tracks_offset, timestamp)
+  end
+
+  defp longer_than_treshold?(%{"offset" => offset} = track, timestamp) do
+    ReportParser.calculate_track_end(track, offset) - timestamp > @avatar_threshold_ns
+  end
+
+  defp has_video_in_threshold?(origin, video_tracks_offset, timestamp) do
+    threshold = timestamp + @avatar_threshold_ns
+
+    next_video_offset =
+      video_tracks_offset
+      |> Map.get(origin, [])
+      |> Enum.find(threshold, &(&1 > timestamp))
+
+    next_video_offset < threshold
+  end
```

Comment:
So, the goal of this function is to handle a situation when someone turns off a camera for a less-than-threshold and we don't want to show an avatar in that case?